osu.Game.Rulesets.Osu/Objects/Drawables/DrawableOsuBlinds.cs

```diff
@@ -130,6 +208,64 @@ protected override void Update()
             panelRight.X = end - width;
             bgPanelLeft.X = start;
             bgPanelRight.X = end;
+
+            float adjustedNpcPosition = npcPosition * rawWidth;
+            if (randomNpc != null)
+                randomNpc.X = adjustedNpcPosition;
+            bgRandomNpc.X = adjustedNpcPosition;
+        }
+
+        public void TriggerNPC()
+        {
```

Comment:
The npc should not behave randomly. Please make it dependend by something like currently time. (or something else constant.)